doc/tutorial/debug_faq.txt

```diff
@@ -295,6 +295,23 @@ Tips:
   of type *float64*.
 
 
+"Why is my GPU function seems to be slow?"
+------------------------------------
+
+When you compile a theano function, if you do not get the speedup that you expect over the 
+CPU performance of the same code. It is oftentimes due to the fact that some Ops might be running
+on CPU instead GPU. If that is the case, you can use assert_no_cpu_op to check if there 
+is a CPU Op on your computational graph. assert_no_cpu_op can take the following one of the three 
+options:
+
+* ``warn``: Raise a warning
+* ``pdb``: Stop with a pdb in the computational graph during the compilation
+* ``raise``: Raise an error,
+if there is a CPU Op in the computational graph.
+
+It is possible to use this mode by providing the flag in THEANO_FLAGS, such as:
+``THEANO_FLAGS="float32,assert_no_cpu_op='raise'" python test.py``
```

Comment:
add device=gpu. Tell that we don't always print them, and I think we still miss some detection pattern.